docs/quickstart.md

```diff
@@ -293,6 +293,511 @@ hive>
 {% include note.html content="This is only supported for Read-optimized tables for now." %}
 
 
+## Testing Hoodie in Local Docker environment
 
+You can bring up a hadoop docker environment containing Hadoop, Hive and Spark services with support for hoodie.
+```
+$ mvn pre-integration-test -DskipTests
+```
+The above command builds docker images for all the services with 
+current hoodie source installed at /var/hoodie/ws and also brings up the services using a compose file. We 
+currently use Hadoop (v2.8.4), Hive (v2.3.3) and Spark (v2.3.1) in docker images. 
+
+To bring down the containers
+```
+$ cd hoodie-integ-test
+$ mvn docker-compose:down
+```
+
+If you want to bring up the docker containers, use
+```
+$ cd hoodie-integ-test
+$  mvn docker-compose:up -DdetachedMode=true
+```
+
+Hoodie is a library that is operated in a broader data analytics/ingestion environment 
+involving Hadoop, Hive and Spark. Interoperability with all these systems is a key objective for us. We are
+actively adding integration-tests under __hoodie-integ-test/src/test/java__ that makes use of this 
+docker environment (See __hoodie-integ-test/src/test/java/com/uber/hoodie/integ/ITTestHoodieSanity.java__ )
+
+
+## A Demo using docker containers
+
+Lets use a real world example to see how hudi works end to end. For this purpose, a self contained
+data infrastructure is brought up in a local docker cluster within your computer. 
+
+The steps assume you are using Mac laptop
+
+### Prerequisites
+
+  * Docker Setup :  For Mac, Please follow the steps as defined in [https://docs.docker.com/v17.12/docker-for-mac/install/]
+  * kafkacat : A command-line utility to publish/consume from kafka topics. Use `brew install kafkacat` to install kafkacat
+  * /etc/hosts : The demo references many services running in container by the hostname. Add the following settings to /etc/hosts
+  
+  ```
+   127.0.0.1 namenode
+   127.0.0.1 datanode1
+   127.0.0.1 hiveserver
+   127.0.0.1 hivemetastore
+   127.0.0.1 kafkabroker
+   127.0.0.1 zookeeper
+  ```
+
+### Setting up Docker Cluster
+
+
+#### Build Hoodie
+
+The first step is to build hoodie
+```
+cd <HUDI_WORKSPACE>
+mvn package -DskipTests
+```
+
+#### Bringing up Demo Cluster
 
+The next step is to run the docker compose script and setup configs for bringing up the cluster.
+This should pull the docker images from docker hub and setup docker cluster.
+
+```
+cd docker
+./setup_demo.sh
```

Comment:
I've encountered this. It seems it's related docker version. Here's my env to figure out problems. 

* docker-engine: 18.06.1-ce (there's no way to downgrade 17.12 since the official site only support latest version for Mac)
* docker-compose: 1.22.0

```
➜  docker git:(docker) ✗ ./setup_demo.sh
Removing network compose_default
Creating network "compose_default" with the default driver
Creating kafkabroker ...
Creating zookeeper                 ... error
Creating hive-metastore-postgresql ...
Creating namenode                  ...
Creating kafkabroker               ... error
ERROR: for zookeeper  Cannot create container for service zookeeper: Conflict. The container name "/zookeeper" is already in use by container "8fb174b9eb8c40bdc20e0ed1b042a793fe49ec3dcea1f353c250b482c7c80019". You have to remove (or rename) that container to be able to reuse that name.

ERROR: for kafkabroker  Cannot create container for service kafka: Conflict. The container name "/kafkabroker" is alreCreating hive-metastore-postgresql ... error
ame) that container to be able to reuse that name.
```

Reply:
@sungjuly : ok, can you check "docker container ls" and see if there are stopped/running containers with the same name. If you find them, can you remove the containers using "docker rm". 

This is an one-time issue for those who tried this PR before. I have moved the compose script to a different file but the container names are same. Removing the docker containers should hopefully fix the issue. 

For a different reason, I reverted my docker (mac) installation to "factory defaults" and that also fixed the problem.

Reply:
docker env was cleaned up all things before I tested. Here's output for `docker container ls`  

```
➜  docker git:(docker) ✗ docker container ls
CONTAINER ID        IMAGE               COMMAND             CREATED             STATUS              PORTS               NAMES
```